src/main/java/de/leonhard/storage/internal/provider/LightningProviders.java

```diff
@@ -24,44 +24,39 @@
   private ExceptionHandler exceptionHandler;
 
   public MapProvider mapProvider() {
-
-    if (mapProvider != null) {
-      return mapProvider;
+    if (mapProvider == null) {
+      mapProvider = new MapProvider() {};
     }
-
-    return mapProvider = new MapProvider() {
-    };
+    return mapProvider;
   }
 
   public YamlConfig yamlConfig() {
-    if (yamlConfig != null) {
```

Comment:
Isn't fail fast a bit cleaner here? But no blocking issue from my perspective.

Reply:
My main issue was the assignment and return in the same statement - also a good practice is to have 1 return statement in a method so I'm making two improvements in one go as far as I'm aware.

Reply:
> also a good practice is to have 1 return statement in a method so I'm making two improvements in one go as far as I'm aware.

I am **not** a fan from having just one return statement in a method. Because if the method gets long, you can return or 'fail fast'. I think returning as fast as possible helps you to read the code, as you can see a block has ended here (we returned, or failed or whatever) and you don't need to read until the end of the method to see what you could have done at the start.

Reply:
If the method gets complicated, sure. But the "fail fast" here is a null check, I think the impact is miniscule.
On my project this would be the approved way to do it, also SonarLint flagged the previous way as Critical so for now I would suggest mine an improvement.

Reply:
Sonar Lint did not like it because of the assignment and return directly. I don't think it complained about the null check itself.